presto-raptor/src/test/java/com/facebook/presto/raptor/storage/organization/TestShardOrganizationManager.java

```diff
@@ -0,0 +1,197 @@
+import org.skife.jdbi.v2.DBI;
+import org.skife.jdbi.v2.Handle;
+import org.skife.jdbi.v2.IDBI;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.OptionalLong;
+import java.util.Set;
+import java.util.UUID;
+
+import static com.facebook.presto.raptor.metadata.SchemaDaoUtil.createTablesWithRetry;
+import static com.facebook.presto.raptor.metadata.TestDatabaseShardManager.createShardManager;
+import static com.facebook.presto.raptor.storage.organization.ShardOrganizationManager.createOrganizationSets;
+import static com.facebook.presto.raptor.storage.organization.TestCompactionSetCreator.extractIndexes;
+import static com.facebook.presto.raptor.storage.organization.TestShardOrganizer.createShardOrganizer;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.DateType.DATE;
+import static com.facebook.presto.spi.type.TimestampType.TIMESTAMP;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.airlift.units.Duration.valueOf;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.stream.Collectors.toSet;
+import static org.testng.Assert.assertEquals;
+
+@Test(singleThreaded = true)
+public class TestShardOrganizationManager
+{
+    private IDBI dbi;
+    private Handle dummyHandle;
+    private MetadataDao metadataDao;
+    private ShardOrganizerDao organizerDao;
+
+    private static final Table tableInfo = new Table(1L, OptionalLong.empty(), OptionalInt.empty(), OptionalLong.empty());
+    private static final Table temporalTableInfo = new Table(1L, OptionalLong.empty(), OptionalInt.empty(), OptionalLong.of(1));
+
+    private static final List<Type> types = ImmutableList.of(BIGINT, VARCHAR, DATE, TIMESTAMP);
+
+    @BeforeMethod
+    public void setup()
+    {
+        dbi = new DBI("jdbc:h2:mem:test" + System.nanoTime());
+        dummyHandle = dbi.open();
+        metadataDao = dbi.onDemand(MetadataDao.class);
+        organizerDao = dbi.onDemand(ShardOrganizerDao.class);
+
+        createTablesWithRetry(dbi);
+    }
+
+    @AfterMethod
+    public void teardown()
+    {
+        dummyHandle.close();
+    }
+
+    @Test
+    public void testOrganizationEligibleTables()
+            throws Exception
+    {
+        long table1 = metadataDao.insertTable("schema", "table1", false, true, null);
+        metadataDao.insertColumn(table1, 1, "foo", 1, "bigint", 1, null);
+
+        metadataDao.insertTable("schema", "table2", false, true, null);
+        metadataDao.insertTable("schema", "table3", false, false, null);
+        assertEquals(metadataDao.getOrganizationEligibleTables(), ImmutableSet.of(table1));
+    }
+
+    @Test(timeOut = 1_000)
```

Comment:
This test might fail on Travis or under other conditions where threads are not scheduled in a timely manner.